test/test_dispatch.py

```diff
@@ -374,6 +374,114 @@ def test_computed_table_with_cpu_catchall(self):
 AutogradXLA: default_def_name_t_t [catch all]
+''')
+
+    def test_computed_table_with_math(self):
+        global_m = C._dispatch_library("IMPL", "_", "autogradcpu")
+        result = self.commute("foo", [
+            # m.def("foo", [](const Tensor & x) { return x })
+            lambda m: m.def_name_t_t("foo"),
+            # m.impl("foo", torch::kMath, [](const Tensor & x) { return x })
+            lambda m: m.impl_t_t("foo", "math"),
+        ])
+        state, table = result.state, result.table
+        self.assertExpectedInline(state, '''\
+name: test::foo
+schema: test::foo(Tensor _0) -> (Tensor _0)
+debug: registered at /dev/null:0
+alias analysis kind: CONSERVATIVE
+Math[alias]: impl_t_t :: (Tensor _0) -> (Tensor _0) [ boxed unboxed ]
+catchall: default_def_name_t_t :: (Tensor _0) -> (Tensor _0) [ boxed unboxed ]
+''')
+
+        # computed dispatch table is too big, so we only check on a few entries we're interested in.
+        extracted_table = extract_dispatch_table_with_keys(
+            table,
+            ('CPU', 'CUDA', 'XLA', 'AutogradOther', 'AutogradCPU', 'AutogradCUDA', 'AutogradXLA'))
+
+        self.assertExpectedInline(extracted_table, '''\
+CPU: impl_t_t [math kernel]
+CUDA: impl_t_t [math kernel]
+XLA: impl_t_t [math kernel]
+AutogradOther: impl_t_t [math kernel]
+AutogradCPU: impl_t_t [math kernel]
+AutogradCUDA: impl_t_t [math kernel]
+AutogradXLA: impl_t_t [math kernel]
+''')
+
+    def test_computed_table_with_cpu_math(self):
+        global_m = C._dispatch_library("IMPL", "_", "autogradcpu")
+        result = self.commute("foo", [
+            # m.def("foo", [](const Tensor & x) { return x })
+            lambda m: m.def_name_t_t("foo"),
+            # m.impl("foo", torch::kCPU, [](const Tensor & x) { return x })
+            lambda m: m.impl_t_t("foo", "cpu", debug="fn1"),
+            # m.impl("foo", torch::kMath, [](const Tensor & x) { return x })
+            lambda m: m.impl_t_t("foo", "math", debug="fn2"),
+        ])
+        state, table = result.state, result.table
+        self.assertExpectedInline(state, '''\
+name: test::foo
+schema: test::foo(Tensor _0) -> (Tensor _0)
+debug: registered at /dev/null:0
+alias analysis kind: CONSERVATIVE
+CPU: fn1 :: (Tensor _0) -> (Tensor _0) [ boxed unboxed ]
+Math[alias]: fn2 :: (Tensor _0) -> (Tensor _0) [ boxed unboxed ]
+catchall: default_def_name_t_t :: (Tensor _0) -> (Tensor _0) [ boxed unboxed ]
+''')
+
+        # computed dispatch table is too big, so we only check on a few entries we're interested in.
+        extracted_table = extract_dispatch_table_with_keys(
+            table,
+            ('CPU', 'CUDA', 'XLA', 'AutogradOther', 'AutogradCPU', 'AutogradCUDA', 'AutogradXLA', 'Meta'))
+
+        self.assertExpectedInline(extracted_table, '''\
+CPU: fn1 [kernel]
+CUDA: fn2 [math kernel]
+XLA: fn2 [math kernel]
+AutogradOther: fn2 [math kernel]
+AutogradCPU: fallthrough registered at /dev/null:0 [backend fallback]
```

Comment:
Huh, where did this fallthrough come from? I don't see any fallthroughs in this test. Is this coming from one of our app-wide registrations and you're munging this? (I must have missed that in post review). If so, it would probably be good to be a little more creative with the file names here so we can easily distinguish between "things that came from this test suite" and "things that came from the framework" (even if we don't want to report their exact line numbers, since those are prone to wobbling.)

Reply:
And I guess, in the terminal state, this is going to be a backend fallback saying that backwards is not supported, right? Not just a fallthrough.

Reply:
Yup this won't be a fallthrough in the terminal state.